test/integration/singlecluster/controller/jobs/job/job_controller_test.go

```diff
@@ -2115,34 +2121,122 @@ var _ = ginkgo.Describe("Job controller interacting with Workload controller whe
 		fwk.StopManager(ctx)
 	})
 
-	ginkgo.When("long backoffBaseSeconds", func() {
+	ginkgo.When("long backoffBaseSeconds and tiny waitForPodsReady.timeout", func() {
 		ginkgo.BeforeEach(func() {
 			backoffBaseSeconds = 10
+			waitForPodsReadyTimeout = &metav1.Duration{Duration: util.TinyTimeout}
```

Comment:
What is the purpose of this change? It is completely non-obvious to me how this is related to the recovery timeout. Please revert if possible. As posted in the other comment, I think it is better to have a bit more of extra code but have a dedicated configuraiton.

Reply:
Those changes were actually needed, as previously for every `When` scenario the waitForPodsReadyTimeout was hardcoded to tiny, as those test check if the Workload is evicted. Given that I wanted to test eviction because of `.recoveryTimeout` and not `.timeout` I had to prolong it, and this is where this change is coming from

Reply:
> Plus, if you don't modify configuration for the existing ones you can test what happens if the pod fails, but there is no recovertyTimeoutSet, which actually is quite important to check.

If we want to test this then the changes need to stay

Reply:
Ah, I got it, so you are not changing the configuration. Just setting it to tiny as it was hardcoded before, fine.